index.js

```diff
@@ -191,59 +196,49 @@ function fastifyView (fastify, opts, next) {
   }
   // Gets template as string (or precompiled for Handlebars)
   // from LRU cache or filesystem.
-  const getTemplate = function (file, callback, requestedPath) {
+  const getTemplate = async function (file, requestedPath) {
     const data = lru.get(file)
     if (data && prod) {
-      callback(null, data)
+      return data
     } else {
-      readFile(join(templatesDir, file), 'utf-8', (err, data) => {
-        if (err) {
-          callback(err, null)
-          return
-        }
+      let data = await readFile(join(templatesDir, file), 'utf-8')
 
-        if (useHtmlMinification(globalOptions, requestedPath)) {
-          data = globalOptions.useHtmlMinifier.minify(data, globalOptions.htmlMinifierOptions || {})
-        }
-        if (type === 'handlebars') {
-          data = engine.compile(data)
-        }
-        lru.set(file, data)
-        callback(null, data)
-      })
+      if (useHtmlMinification(globalOptions, requestedPath)) {
+        data = globalOptions.useHtmlMinifier.minify(data, globalOptions.htmlMinifierOptions || {})
+      }
+      if (type === 'handlebars') {
+        data = engine.compile(data)
+      }
+      lru.set(file, data)
+      return data
     }
   }
 
   // Gets partials as collection of strings from LRU cache or filesystem.
-  const getPartials = function (page, { partials, requestedPath }, callback) {
+  const getPartials = async function (page, { partials, requestedPath }) {
     const cacheKey = getPartialsCacheKey(page, partials, requestedPath)
     const partialsObj = lru.get(cacheKey)
+
     if (partialsObj && prod) {
-      callback(null, partialsObj)
+      return partialsObj
     } else {
       let filesToLoad = Object.keys(partials).length
       if (filesToLoad === 0) {
-        callback(null, {})
-        return
+        return {}
       }
-      let error = null
       const partialsHtml = {}
-      Object.keys(partials).forEach((key, index) => {
-        readFile(join(templatesDir, partials[key]), 'utf-8', (err, data) => {
-          if (err) {
-            error = err
-          }
-          if (useHtmlMinification(globalOptions, requestedPath)) {
-            data = globalOptions.useHtmlMinifier.minify(data, globalOptions.htmlMinifierOptions || {})
-          }
 
-          partialsHtml[key] = data
-          if (--filesToLoad === 0) {
-            lru.set(cacheKey, partialsHtml)
-            callback(error, partialsHtml)
-          }
-        })
-      })
+      for (const key of Object.keys(partials)) {
```

Comment:
You should use a `Promise.all` here

Reply:
Adding `nextTick` is about error handling, not performance. If you don't, the callback with be called twice in case of an error.

The original code did the `readFile` in parallel, so you should do the same here. with `Promise.all`.

Reply:
I tried to follow all suggestions. Replaced for of with Promise.all().
Maybe in a wrong way. Perfomances with autocannon did not increse.